lib/statsig_logging.ex

```diff
@@ -0,0 +1,123 @@
+defmodule Statsig.Logging do
+  use GenServer
+  require Logger
+
+  @type state :: %{
+    flush_interval: integer() | nil,
+    events: list(),
+    flush_timer: reference() | nil
+  }
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, %{
+      flush_interval: 60_000,
+      events: [],
+      flush_timer: nil,
+    }, name: __MODULE__)
+  end
+
+  def initialize(flush_interval) do
+    GenServer.call(__MODULE__, {:initialize, flush_interval})
+  end
+
+  def shutdown do
+    GenServer.call(__MODULE__, :shutdown)
+  end
+
+  @impl true
+  def init(state) do
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_call(:shutdown, _from, state) do
+    Logger.info("Shutting down Statsig.Logging")
+
+
+    new_state = if length(state.events) > 0 do
```

Comment:
note: calling length on `state.events` is a O(n) operation. Suggest just calling flush_events without checking.